### PR TITLE
Improve static typing and style

### PR DESCRIPTION
**Pull request by Qubus0**
Lots of little things to make it easier to avoid issues in the future.
Please test this with your mods, it should not have any effect at all.
Let me know if I missed something

### CODE REVIEW

File: addons/mod_loader/mod_loader.gd [hunk 1]
```diff
@@ -497,37 +493,37 @@ func save_scene(modified_scene, scene_path:String):
...
+	var error_msg := ""
+	var data = {} # can be anything
+	var defaults := {}
 
 	# Invalid mod ID
-	if !mod_data.has(mod_id):
+	if not mod_data.has(mod_dir_name):
 		error_num = 1
-		error_msg = str("ERROR - Mod ID was invalid: ", mod_id)
+		error_msg = "ERROR - Mod ID was invalid: " % mod_dir_name
```
Comment (otDan) [inline addons/mod_loader/mod_loader.gd:505]: missing from string

Comment (Qubus0) [reply addons/mod_loader/mod_loader.gd]: good catch! fixed

**Comment by KANAjetzt** [comment]
I think #70 is missing on this one - right?

**Comment by Qubus0** [comment]
@KANAjetzt develop is missing #70 as well, since it got merged past

**Review by ithinkandicode — APPROVED** [review]
👍

**Pull request merged**